viewer/Viewer.h

```diff
@@ -111,23 +124,58 @@ namespace Enki
 		typedef QMapIterator<const std::type_info*, const std::type_info*> ManagedObjectsAliasesMapIterator;
 		ManagedObjectsAliasesMap managedObjectsAliases;
 		
+		typedef std::pair<QString, unsigned int> viewerMessage;
+		std::list<viewerMessage> messageList;
+		QString controlError1;	// dislpayed if user try to move an object in trackball mode : action forbiden to prevent glitchies
+		QString controlError2;	// dislpayed if user try to move camera in trackball mode
+		QString controlHelp;	// dislpayed if user press F1
+
+		struct ExtendedAttributes
+		{
+			bool movableByPicking;
+
+			ExtendedAttributes():movableByPicking(false){};
+		};
+		std::map<PhysicalObject*, ExtendedAttributes> objectExtendedAttributesList;
+
 		bool mouseGrabbed;
 		QPoint mouseGrabPos;
 		double wallsHeight;
-		
-		bool doDumpFrames;
+		//! to know if camera is in trackball mode
+		bool trackballView;
 		int dumpFramesCounter;
 	
+		PhysicalObject *pointedObject, *selectedObject;
+		QVector3D pointedPoint;
+
 	public:
 		ViewerWidget(World *world, QWidget *parent = 0);
 		~ViewerWidget();
 	
+		void addManagedObjectsAlias(const std::type_info* key, const std::type_info* value);
+
+		World* getWorld();
+		QVector3D getPointedPoint();
+		PhysicalObject* getPointedObject();
+		PhysicalObject* getSelectedObject();
+		bool isTrackballActivated();
+		QString getHelpString();
+		bool isMovableByPicking(PhysicalObject* object);
+
+		void setMovableByPicking(PhysicalObject* object, bool movable);
+
+		void timerEvent(QTimerEvent * event);
```

Comment:
It's better to have the application (playground) update in the same time of the viewer update, so I choose to use a single timerEvent function in the application level and call the viewer one in the application one (better synchronization to me),
maybe I can call the simulation update and after the GL update instead of this two line function to be more clear (the name timerEvent can be misleading because it's not a regular "qt slot")